lib/config/default.js

```diff
@@ -704,29 +841,31 @@ exports.config = () => ({
      * Relic servers. The memory concerns are something you should consider for
      * your own server's sake. The payload of events is compressed, but if it
      * grows too large the New Relic servers may reject it.
-     *
-     * @env NEW_RELIC_CUSTOM_INSIGHTS_EVENTS_MAX_SAMPLES_STORED
      */
-    max_samples_stored: 3000
+    max_samples_stored: {
+      formatter: int,
+      default: 3000
+    }
   },
   /**
    * This is used to configure properties about the user's host name.
    */
   process_host: {
     /**
      * Configurable display name for hosts
-     *
-     * @env NEW_RELIC_PROCESS_HOST_DISPLAY_NAME
      */
     display_name: '',
     /**
      * ip address preference when creating hostnames
-     *
-     * @env NEW_RELIC_IPV_PREFERENCE
      */
-    ipv_preference: '4'
+    ipv_preference: {
+      formatter(val) {
+        return ['4', '6'].includes(val) ? val : '4'
```

Comment:
It seems like it would be useful to have an "allow list" formatter, since this and record_sql use it, and I could see the use-case in the future